charms/knative-operator/src/charm.py

```diff
@@ -187,18 +233,38 @@ def _apply_resources(self, resource_handler):
 
     def _main(self, event):
         """Event handler for changing Pebble configuration and applying k8s resources."""
-        # Update Pebble configuration layer if it has changed
-        self.unit.status = MaintenanceStatus("Configuring Pebble layer")
-        self._update_layer(event)
-
         # Apply Kubernetes resources
         self.unit.status = MaintenanceStatus("Applying resources")
         self._apply_resources(resource_handler=self.resource_handler)
 
+        # TODO: There is a race condition between creating the configmaps and secrets that
+        #  knative operator/webhook need, and starting their processes.  Without these resources,
+        #  their processes panic and exit with non-zero error codes quickly (<1s).  Pebble sees
+        #  this quick exit not as a start and fail, not as not starting at all (see
+        #  https://github.com/canonical/pebble#viewing-starting-and-stopping-services), which
+        #  means Pebble will not auto-restart the service.  Health checks also do not seem to
+        #  be able to restart the service.
+        #  Need a better way to handle this.
+        sleep_time = 5
```

Comment:
Documented this as #90 and (if you agree) fixed it now in this PR by adding `wait_for_required_kubernetes_resources()`

Reply:
Adding `_on_knative_operator[_webhook]_pebble_ready()` still won't ensure the resources are there before the service is started by pebble in a pebble ready event, you need something else. I am not 100% in favor of removing the pebble ready event handlers because it goes against the design of this charm, but it is a solution for now. #90 is good enough for continuing this discussion.